lib/improver/tests/set_up_test_cubes.py

```diff
@@ -0,0 +1,331 @@
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""
+Functions to set up variable, multi-realization, percentile and probability
+cubes for unit tests.  Standardises time units and spatial coordinates,
+including coordinate order expected by IMPROVER plugins.
+"""
+
+from cf_units import Unit, date2num
+from datetime import datetime
+
+import numpy as np
+import iris
+from iris.coords import DimCoord
+
+from improver.utilities.cube_checker import check_cube_not_float64
+
+TIME_UNIT = "seconds since 1970-01-01 00:00:00"
+CALENDAR = "gregorian"
+
+
+def construct_xy_coords(ypoints, xpoints, spatial_grid):
+    """
+    Construct x/y spatial dimension coordinates
+
+    Args:
+        ypoints (int):
+            Number of grid points required along the y-axis
+        xpoints (int):
+            Number of grid points required along the x-axis
+        spatial_grid (str):
+            Specifier to produce either a lat-lon or equal area grid
+
+    Returns:
+        y_coord, x_coord (tuple):
+            Tuple of iris.coord.DimCoord instances
+    """
+    if spatial_grid == 'latlon':
+        # make a lat-lon grid including the UK area
+        y_coord = DimCoord(
+            np.linspace(-20.0, 20.0, ypoints, dtype=np.float32),
+            "latitude", units="degrees")
+        x_coord = DimCoord(
+            np.linspace(40, 80, xpoints, dtype=np.float32),
+            "longitude", units="degrees")
+    else:
+        # use UK eastings and northings
+        y_coord = DimCoord(
+            np.linspace(-100000, 900000, ypoints, dtype=np.float32),
+            "projection_y_coordinate", units="metres")
+        x_coord = DimCoord(
+            np.linspace(-400000, 600000, xpoints, dtype=np.float32),
+            "projection_x_coordinate", units="metres")
+    return y_coord, x_coord
```

Comment:
The coordinates returned by this function do not include a defined coordinate system. This is important for lots of functions to do with regridding. If we want to be able to use this cube maker for testing that kind of functionality, we need a defined coordinate system on the cube/coordinates.

Reply:
Used STANDARD_GRID_CCRS when `spatial_grid="equalarea"` is specified.